src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -272,6 +281,12 @@ public ResponseEntity<ProgrammingExercise> updateProgrammingExercise(@RequestBod
         // Forbid conversion between normal course exercise and exam exercise
         exerciseService.checkForConversionBetweenExamAndCourseExercise(updatedProgrammingExercise, programmingExerciseBeforeUpdate, ENTITY_NAME);
 
+        // Check that only allowed Athena modules are used
+        athenaModuleService.ifPresentOrElse(ams -> ams.checkHasAccessToAthenaModule(updatedProgrammingExercise, course, ENTITY_NAME),
+                () -> updatedProgrammingExercise.setFeedbackSuggestionModule(null));
+        // Changing Athena module after the due date has passed is not allowed
```

Comment:
This check seems to be duplicated multiple times among the codebase. Maybe a routine that does this allowed athena modules check would reduce code duplication.

Reply:
Will be also refactored in a follow up